File: app/src/main/java/taxi/kassa/model/responses/Order.java
package taxi.kassa.model.responses;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Order {

    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("source_id")
    @Expose
    private String sourceId;
    @SerializedName("address_from")
    @Expose
    private String addressFrom;
    @SerializedName("address_to")
    @Expose
    private String addressTo;
    @SerializedName("amount_client")
    @Expose
    private String amountClient;
    @SerializedName("amount_driver")
    @Expose
    private String amountDriver;
    @SerializedName("amount_tip")
    @Expose
    private String amountTip;
    @SerializedName("amount_fee_agr")
    @Expose
    private String amountFeeAgr;
    @SerializedName("amount_fee_our")
    @Expose
    private String amountFeeOur;
    @SerializedName("amount_total")
    @Expose
    private Float amountTotal;
    @SerializedName("created")
    @Expose
    private String created;
    @SerializedName("statunt_total")
    @Expose
    private Float statuntTotal;
    @SerializedName("ant_tip")
    @Expose
    private String antTip;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getSourceId() {
        return sourceId;
    }

    public void setSourceId(String sourceId) {
        this.sourceId = sourceId;
    }

    public String getAddressFrom() {
        return addressFrom;
    }

    public void setAddressFrom(String addressFrom) {
        this.addressFrom = addressFrom;
    }

    public String getAddressTo() {
        return addressTo;
    }

    public void setAddressTo(String addressTo) {
        this.addressTo = addressTo;
    }

    public String getAmountClient() {
        return amountClient;
    }

    public void setAmountClient(String amountClient) {
        this.amountClient = amountClient;
    }

    public String getAmountDriver() {
        return amountDriver;
    }

    public void setAmountDriver(String amountDriver) {
        this.amountDriver = amountDriver;
    }

    public String getAmountTip() {
        return amountTip;
    }

    public void setAmountTip(String amountTip) {
        this.amountTip = amountTip;
    }

    public String getAmountFeeAgr() {
        return amountFeeAgr;
    }

    public void setAmountFeeAgr(String amountFeeAgr) {
        this.amountFeeAgr = amountFeeAgr;
    }

    public String getAmountFeeOur() {
        return amountFeeOur;
    }

    public void setAmountFeeOur(String amountFeeOur) {
        this.amountFeeOur = amountFeeOur;
    }

    public Float getAmountTotal() {
        return amountTotal;
    }

    public void setAmountTotal(Float amountTotal) {
        this.amountTotal = amountTotal;
    }

    public String getCreated() {
        return created;
    }

    public void setCreated(String created) {
        this.created = created;
    }

    public Float getStatuntTotal() {
        return statuntTotal;
    }

    public void setStatuntTotal(Float statuntTotal) {
        this.statuntTotal = statuntTotal;
    }

    public String getAntTip() {
        return antTip;
    }

    public void setAntTip(String antTip) {
        this.antTip = antTip;
    }

}

File: app/src/main/java/taxi/kassa/model/responses/Order.kt
package taxi.kassa.model.responses

import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import taxi.kassa.util.Constants.DAY_YEAR_PATTERN
import taxi.kassa.util.Constants.FULL_PATTERN
import taxi.kassa.util.Constants.HOURS_PATTERN
import taxi.kassa.util.Constants.myDateFormatSymbols
import java.text.SimpleDateFormat
import java.util.*

@Parcelize
data class Order(
    @SerializedName("id")
    @Expose
    var id: String? = null,
    @SerializedName("status")
    @Expose
    var status: String? = null,
    @SerializedName("source_id")
    @Expose
    var sourceId: String? = null,
    @SerializedName("address_from")
    @Expose
    var addressFrom: String? = null,
    @SerializedName("address_to")
    @Expose
    var addressTo: String? = null,
    @SerializedName("amount_client")
    @Expose
    var amountClient: String? = null,
    @SerializedName("amount_driver")
    @Expose
    var amountDriver: String? = null,
    @SerializedName("amount_tip")
    @Expose
    var amountTip: String? = null,
    @SerializedName("amount_fee_agr")
    @Expose
    var amountFeeAgr: String? = null,
    @SerializedName("amount_fee_our")
    @Expose
    var amountFeeOur: String? = null,
    @SerializedName("amount_total")
    @Expose
    var amountTotal: Float? = null,
    @SerializedName("created")
    @Expose
    var created: String? = null,
    @SerializedName("statunt_total")
    @Expose
    var statuntTotal: Float? = null,
    @SerializedName("ant_tip")
    @Expose
    var antTip: String? = null
) : Parcelable {

    val date: String
        get() {
            val longDate = (created?.toInt()?.toLong() ?: 0) * 1000
            val date = Date(longDate)
            val dateFormat = SimpleDateFormat("dd.MM.yy")
            dateFormat.timeZone = TimeZone.getDefault()

            return dateFormat.format(date)
        }

    val hours: String
        get() {
            val longDate = (created?.toInt()?.toLong() ?: 0) * 1000
            val date = Date(longDate)
            val dateFormat = SimpleDateFormat(HOURS_PATTERN)
            dateFormat.timeZone = TimeZone.getDefault()

            return dateFormat.format(date)
        }

    fun getDateForTitle(): String {
        val longDate = (created?.toInt()?.toLong() ?: 0) * 1000
        val date = Date(longDate)
        val dateFormat = SimpleDateFormat(DAY_YEAR_PATTERN, myDateFormatSymbols)
        dateFormat.timeZone = TimeZone.getDefault()

        return dateFormat.format(date)
    }

    fun getDateWithTime(): String {
        val longDate = (created?.toInt()?.toLong() ?: 0) * 1000
        val date = Date(longDate)
        val dateFormat = SimpleDateFormat(FULL_PATTERN, myDateFormatSymbols)
        dateFormat.timeZone = TimeZone.getDefault()

        return dateFormat.format(date)
    }
}